http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -129,17 +130,55 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
         case Some(status) => F.raiseError[Unit](RuntimeAlreadyExistsException(cloudContext, runtimeName, status))
         case None =>
           for {
-            diskToSave <- F.fromEither(
-              convertToDisk(
-                userInfo,
-                cloudContext,
-                DiskName(req.azureDiskConfig.name.value),
-                config.azureConfig.diskConfig,
-                req,
-                landingZoneResources.region,
-                ctx.now
-              )
-            )
+            diskId <- useExistingDisk match {
+              // if using existing disk, find disk in users workspace
+              case true =>
+                for {
+                  disks <- DiskServiceDbQueries
+                    .listDisks(Map.empty, includeDeleted = false, Some(userInfo.userEmail), Some(cloudContext))
+                    .transaction
+                  // check if 0 or multiple disks
+                  disk <- disks.length match {
+                    case 1 => F.pure(disks.head)
+                    case 0 => F.raiseError(NoPersistentDiskException(workspaceId))
+                    case _ =>
+                      F.raiseError(MultiplePersistentDisksException(workspaceId, disks.length, disks))
+                  }
+                  // check disk is ready
+                  _ <- F
+                    .raiseError(PersistentDiskNotReadyException(disk.id, disk.status))
+                    .whenA(disk.status != DiskStatus.Ready)
+
+                  // check previously attached runtime is deleted
+                  runtimeOpt <- clusterQuery.getClusterFromDiskId(disk.id).transaction
+                  _ <- F
+                    .raiseError(
+                      new Exception(
+                        s"This should never happen. No previous runtime associated with disk ${disk.id}"
+                      )
+                    )
+                    .whenA(runtimeOpt.isEmpty)
+                  _ <- F
+                    .raiseError(NoUnattachedPersistentDiskException(disk.id, runtimeOpt.get.runtimeName))
+                    .whenA(runtimeOpt.get.status != RuntimeStatus.Deleted)
+                  id = disk.id
+                } yield id
+              case _ =>
```

Comment:
would be nice to use `case false` here, and add a comment to explain what this branch is doing. Always helps when nesting complex `for` comprehensions